### PR TITLE
Fix order specific mount.mounted options for persist

### LINKED ISSUE #62556: [BUG] mount.mounted options are order specific for persist
**Description**
Similar to (but not the same as) #19003 and #55769

Essentially, mount options provided to the state are natural sort ordered and expected to be in that order in the persisted mount file (such as `/etc/fstab`). This is problematic for writing states to take over management of existing mounts because a state cannot be written in a way that will run "clean" if the options order is not sorted in fstab.

**Setup**

```yaml
# /srv/salt/mount.sls

alpine:
  mount.mounted:
    - name: /mnt/iso
    - device: /srv/salt/alpine-standard-3.16.2-x86_64.iso
    - fstype: iso9660
    - opts:
      - auto
      - loop
      - ro
    - persist: True
    - mkmnt: True
```

Running this state shows that changes will be made even though the mount is present with all options present:

```
# /etc/fstab
...
/srv/salt/alpine-standard-3.16.2-x86_64.iso             /mnt/iso        iso9660 auto,ro,loop    0 0
```

**Expected behavior**
I would expect that the options would be compared irrespective of order, since order of the options is not important to mount.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```yaml
Salt Version:
          Salt: 3005+0na.c57a277
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.4
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: 2.6.1
  pycryptodome: 3.15.0
        pygit2: Not Installed
        Python: 3.7.3 (default, Jan 22 2021, 20:04:44)
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 23.2.1
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: debian 10 buster
        locale: UTF-8
       machine: x86_64
       release: 4.19.0-21-amd64
        system: Linux
       version: Debian GNU/Linux 10 buster

```
</details>

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
This PR implements similar functionality to what was proposed by @thusoy in #57506, and does not restrict mount option comparisons to the same order.

### What issues does this PR fix or reference?
Fixes: #62556

### Previous Behavior
See issue for details.

### New Behavior
Mount options are compared irrespective of order in the file and state block. This allows for comparison of functionality instead of literal string matching.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by nicholasmhughes** [comment]
re-run pr-amazon-2-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-amazon-2-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-amazon-2-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-amazon-2-x86_64-py3-pytest

File: salt/modules/mount.py [hunk 1]
```diff
@@ -867,6 +882,9 @@ def set_fstab(
                 line = salt.utils.stringutils.to_unicode(line)
                 try:
                     if criteria.match(line):
+                        log.debug(
+                            "Checking ( %s ) against current line ( %s )", entry, line
```
Comment (Ch3LL) [inline salt/modules/mount.py:886]: If someone has credentials in their fstab file this would log the password I believe.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/mount.py (hunk 1 above)
Comment (nicholasmhughes) [reply salt/modules/mount.py]: removed

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**